File: vison/hist.py
import cv2
import matplotlib.pyplot as plt
#  Load the image
img = cv2.imread('sample.jpg')
img = cv2.resize(img,(250,350))
gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
cv2.imshow('gray',gray)

# Calculate Histogram use matpl1otlib
plt.hist(img.ravel(),256,[0,256])
plt.show()

# creating a Histograms Equalization use OpenCV
equ = cv2.equalizeHist(gray) 
cv2.imshow('image', equ)

plt.hist(equ.ravel(),256,[0,256])
plt.show()

File: vison/hist.m
clc; clear all; close all;
im=imread('sample.jpg');
g=rgb2gray(im);
imshow(g)
g2=im2double(g);
J1=1*(g2);
J2=2.5*(g2);
J3=0.5*(g2);
figure, imshow(J1)
figure, imshow(J2)
figure, imshow(J3)
